examples/with-next/pages/index.tsx

```diff
@@ -193,69 +190,61 @@ export default class Index extends PureComponent {
-      'five',
-      'six',
-      'seven',
-      'eight',
-      'nine'
-    ]
-    const config: PipelineConfig = {
-      onReady: () => {
-        this.setState({ keyword: { status: 'Listening', result: '' } })
-      },
-      onEvent: (evt) => {
-        const { eventType, transcript } = evt
-        if (eventType == EventType.Recognize) {
-          this.setState({ keyword: { error: '', result: transcript } })
-        } else if (eventType == EventType.Timeout) {
-          this.setState({ keyword: { error: 'timeout' } })
-        }
-      },
-      speechConfig: speechConfig,
-      stages: [Stage.VadTrigger, Stage.KeywordRecognizer]
+    const profile = PipelineProfile.Keyword
+    profile.speechConfig = {
+      ...profile.speechConfig,
+      baseKeywordUrl: 'https://s.spokestack.io/u/UbMeX/js',
+      keywordClasses: [
+        'zero',
+        'one',
+        'two',
+        'three',
+        'four',
+        'five',
+        'six',
+        'seven',
+        'eight',
+        'nine'
+      ]
+    }
+    profile.onReady = () => {
+      this.setState({ keyword: { status: 'Listening', result: '' } })
     }
-    this.pipeline = new Pipeline(config)
-    this.pipeline.listen()
+    profile.onEvent = (evt) => {
+      const { eventType, transcript } = evt
+      if (eventType == EventType.Recognize) {
+        this.setState({ keyword: { error: '', result: transcript } })
+      } else if (eventType == EventType.Timeout) {
+        this.setState({ keyword: { error: 'timeout' } })
+      }
+    }
+
+    startPipeline(profile)
   }
 
   startWakeword = async () => {
     this.initialize()
     this.setState({ wakeword: { status: 'Calibrating...' } })
-    const speechConfig: SpeechConfig = DefaultSpeechConfig
-    speechConfig.baseWakewordUrl = 'https://s.spokestack.io/u/hgmYb/js'
-    const config: PipelineConfig = {
-      onReady: () => {
-        this.setState({ wakeword: { status: 'Listening', result: '' } })
-      },
-      onEvent: (evt) => {
-        const { eventType } = evt
-        if (eventType == EventType.Activate) {
-          this.setState({ wakeword: { error: '', result: true } })
-        } else if (eventType == EventType.Timeout) {
-          this.setState({ wakeword: { error: 'timeout' } })
-        }
-      },
-      speechConfig: speechConfig,
-      stages: [Stage.WakewordTrigger]
+    const profile = PipelineProfile.Wakeword
+    profile.speechConfig.baseWakewordUrl = 'https://s.spokestack.io/u/hgmYb/js'
+    profile.onReady = () => {
+      this.setState({ wakeword: { status: 'Listening', result: '' } })
+    }
+    profile.onEvent = (evt) => {
+      const { eventType } = evt
+      if (eventType == EventType.Activate) {
+        this.setState({ wakeword: { error: '', result: true } })
+      } else if (eventType == EventType.Timeout) {
+        this.setState({ wakeword: { error: 'timeout' } })
+      }
     }
-    this.pipeline = new Pipeline(config)
-    this.pipeline.listen()
+
+    startPipeline(profile)
```

Comment:
I imagined `profile` as a property on the options passed to `startPipeline`. Also, now that I see this, I think we can remove `onReady` and make `startPipeline` return a promise. Like this...

```ts
await startPipeline({
  profile: PipelineProfile.Wakeword,
  baseUrl: 'https://s.spokestack.io/u/hgmYb/js',
  onEvent: (evt) => { ... }
})
this.setState({ wakeword: { status: 'Listening', result: '' } })
```